torchao/experimental/kernels/mps/metal/int3mm_opt.metal

```diff
@@ -0,0 +1,146 @@
+#include <metal_simdgroup>
+#include <metal_stdlib>
+using namespace metal;
+
+inline void unpack_3bit(const uchar3 b, thread float* w) {
+  w[0] = float(((b[0] & 1) << 2) | (b[1] & 3));
```

Comment:
I dont know which of these shifts are avoidable and can be accounted for via act_div_scale stuff done in 4-bit kernel

Reply:
In future PRs I will experiment with different packing schemes, to see which one is optimal. This is the next thing I will do with 3-bit. But I don't want to do that in this PR.
Now, this packing isn't friendly with the act_div_scale stuff. This is because this packing spreads each weight value to multiple bytes. It isn't as clean as 4-bit where every value was contained in a single byte.
I am not saying that we won't use the act_div_scale trick in a future PR. But if you accept that we are not modifying the packing in this PR, then you should be ok with us not trying to complicate this PR with the act_div_scale trick either.

Reply:
I am ok with not complicating this PR. I agree that how values are packed has impact on what you can do.